src/main/java/com/xeeva/navigation/MainNav.java

```diff
@@ -1,309 +1,316 @@
+	@FindBy(id = "lnkShowPopup")	private Link lnkCartItem;
+	@FindBy(linkText = "tblCartInfo")	private Webtable cartTable;
+	@FindBy(xpath = ".//*[@id='tblCartInfo']/tbody/tr")	private List<WebElement> cartItemGridRows;
+	@FindBy(id = "txtQuantity")	private Textbox txtQuantity;
+	@FindBy(id = "ddlUOM")	private Listbox unitOfMeasure;
+	@FindBy(id = "txtUnitPrice")	private Textbox txtUnitPrice;
+	@FindBy(id = "btnSubmit")	private Button btnSubmit;
+	@FindBy(xpath = ".//*[@id='txtItem']")	private Textbox itemDescription;
+	@FindBy(xpath = ".//*[@id='spanCartValue']")	private Element lblCartValue;
+
+
+	// Update Cart Items - for non price agreement cart Items verifications
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[2]/td[2]/input")
+	private Textbox txtItemQuantity;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[1]/td[4]/select")
+	private Listbox lstItemunitOfMeasure;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[1]/td[2]/input[1]")
+	private Textbox txtItemUnitPrice;
+	@FindBy(xpath = ".//*[@id='tblItemDetails']/tbody/tr[2]/td[2]/a/div")
+	private Button btnAddToCart;
+	@FindBy(xpath = ".//*[@value='Save Cart'][contains(@onclick, 'javascript')]") private Button btnsaveCartPopUp;
+	@FindBy(xpath = "//*[@id='btnPopupCP']") private Button btnCheckOut;
+
+	/** Constructor **/
+
+	public MainNav(OrasiDriver driver) {
+		this.driver = driver;
+		pl.isDomComplete(driver);
+		ElementFactory.initElements(driver, this);
+	}
+
+	/** Page Interactions **/
+
+	public void pageLoaded() {
+		lnkLogout.syncVisible(10, false);
+	}
+
+	public boolean isLogoutDisplayed() {
+		try {
+			Thread.sleep(3000);
+		} catch (InterruptedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return lnkLogout.syncVisible(20, false);
+	}
+
+
+	/**
+	 * @summary  Method to click on Cart-Items Link
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+
+	public void clickCartItemsLink() {
+		driver.executeJavaScript("arguments[0].click();", lnkCartItem);
+		driver.setPageTimeout(4);
+	}
+
+	/**
+	 * @summary Method for Application Logout
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void SaveCart() {
+		if (btnSaveCart.syncVisible(1, false)){
+			btnSaveCart.click();
+		} else {
+			TestReporter.logStep("Cart Empty to Save");
+		}
+	}
+
+	/**
+	 * @summary  Method to Click on Cart Item Edit icon
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void click_CartItemEdit(String itemNumber) {
+		List<WebElement> cartRows = driver.findElements(By.xpath(".//*[@id='tblCartInfo']/tbody/tr"));
+		for(int row =1;row<cartRows.size();row++)
+			if(cartRows.get(row).getAttribute("id").contains("trItemRow_")){
+				List<WebElement> links =  cartRows.get(row).findElements(By.tagName("a"));
+				TestReporter.logStep( "Item Number Fronm Cart Rows : "+links.get(0).getText());
+				if(links.get(0).getText().equalsIgnoreCase(itemNumber)){
+					// clicking on Cart Item Edit
+					driver.setPageTimeout(3);
```

Comment:
Remove this setPageTimeout line please.  You can use the setElementTimeout() method instead, but set it back to the default afterwards